crates/amaru-kernel/src/lib.rs

```diff
@@ -799,6 +836,41 @@ impl<'b> HasAddress for PseudoTransactionOutput<MintedPostAlonzoTransactionOutpu
     }
 }
 
+pub trait HasDatum {
+    fn has_datum(&self) -> Option<DatumOption>;
+}
+
+impl HasDatum for TransactionOutput {
+    fn has_datum(&self) -> Option<DatumOption> {
+        match self {
+            PseudoTransactionOutput::Legacy(transaction_output) => {
+                transaction_output.datum_hash.map(DatumOption::Hash)
+            }
+            PseudoTransactionOutput::PostAlonzo(transaction_output) => {
+                // TODO: remove clones
+                transaction_output.datum_option.clone()
+            }
+        }
+    }
+}
+
+impl HasDatum for MintedTransactionOutput<'_> {
+    fn has_datum(&self) -> Option<DatumOption> {
+        match self {
+            PseudoTransactionOutput::Legacy(transaction_output) => {
+                transaction_output.datum_hash.map(DatumOption::Hash)
+            }
+            PseudoTransactionOutput::PostAlonzo(transaction_output) => transaction_output
+                .datum_option
+                .as_ref()
+                .and_then(|datum| match datum {
+                    PseudoDatumOption::Hash(hash) => Some(DatumOption::Hash(*hash)),
+                    PseudoDatumOption::Data(_) => None,
```

Comment:
That's a confusing semantic here. The function returns an `Option<DatumOption>` and is called `has_datum`, but would yield `None` when the underlying datum is actually inline? If that's an oversight, let's fix it. If that's the intended behavior, then that function needs both another signature and a different name.

Seems like it always returns a hash; so we could fancy `-> Option<Hash<32>>` as a return annotation. However, the whole trait implementation is inconsistent with the one for TransactionOutput.

I am confused.

Reply:
You're right, looking back on code I wrote a week ago I got confused 😅 

The idea is this returns the DatumHash, if present, and the `TransactionOutput` implementation is wrong. Will clean this up.